src/mpags-cipher.cpp

```diff
@@ -91,18 +98,36 @@ int main(int argc, char* argv[])
 
     std::string outputText;
 
-    switch (settings.cipherType[0]) {
-        case CipherType::Caesar: {
-            // Run the Caesar cipher (using the specified key and encrypt/decrypt flag) on the input text
-            CaesarCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
+
+    std::vector<std::unique_ptr<Cipher>> cipherVec{};
+    cipherVec.reserve(settings.cipherType.size());
+
+    for (size_t iCipher = 0; iCipher < settings.cipherType.size(); iCipher++) {
+        //std::cout << "iCipher = " << iCipher << std::endl;
+        auto cipher = CipherFactory::makeCipher(settings.cipherType[iCipher], settings.cipherKey[iCipher]);
+        if (!cipher) {
+            std::cerr << "[error] problem constructing cipher" << std::endl;
+            return 1;
         }
-        case CipherType::Playfair: {
-            PlayfairCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
+        cipherVec.push_back(std::move(cipher));
+    }
+
+    switch (settings.cipherMode)
+    {
+        case(CipherMode::Encrypt):
+            for (auto&& cipher : cipherVec) {
+                outputText =  cipher->applyCipher(inputText, settings.cipherMode);
+                inputText = outputText;
+            }
+            break;
+        
+        case(CipherMode::Decrypt):
+            std::reverse(cipherVec.begin(), cipherVec.end());
+            for (auto&& revCipher : cipherVec) {
+                outputText = revCipher->applyCipher(inputText, settings.cipherMode);
+                inputText = outputText;
+            }
```

Comment:
This could be simplified a bit by just doing a conditional to check if the vector should be reversed or not rather than doing a full `switch` statement. In addition, you can just overwrite `inputText` each time and then set `outputText` to it after the loop.